### ISSUE TITLE
No parameter sliders after load

### ISSUE THREAD

**Issue by martblek** [plugdata-team/plugdata#1376]
I don't know if it's a mistake, but if I save a patch with parameters for VST compilation,
then when I upload it later, the parameters don't appear and they need to be defined again.
If I have many @hv_param parameters it is tedious.
thanks

**Comment by dromer** [comment]
Ah you mean the `Automation Parameters` in the plugdata sidebar?

I actually almost never used these myself, but I believe the issue is likely that these are not saved with your pd patch.

They are part of the plugdata state and are only saved in plugin mode together with the entire plugdata state in your DAW session.

There was some discussion to have like a "project mode" in plugdata where such states could be saved/retrieved. Likely you don't want the same parameters for every patch/instance, but only for your specific project.

**Comment by martblek** [comment]
Thanks for answer.
We are talking about compiling a patch to a VST3 plugin from the standalone version of Plugdata.
I have a working patch and I want to export it as a binary VST3 plugin.
So I replaced all sliders with automation parameters. I can't see it now :(

Regarding the use of the Plugdata vst version in the DAW, the parameters are saved there and when loaded,
they are shown and work, but there is another problem, namely that the automation names are not updated.
video[ here](https://youtu.be/tgkH-E-0POg)

**Comment by dromer** [comment]
Yes as I said for standalone the plugdata state is not saved anywhere, so these automation parameters have nowhere to go.

For plugin version this is a different issue, but looks like a bug that could be solved (at least for some plugin versions).

**Comment by martblek** [comment]
Ok thanks.
so we can close it.

**Comment by dromer** [comment]
No don't close the issue. Have @timothyschoen look at these things first :)

**Comment by pereshaped** [comment]
I'm having the same issue. All param automations created vanish after saving, so bassically you cannot use PLugdata as a reliable plugin for musical purposes as you have no way of controlling the PD parameters once you load again. Ive been developing Ekkko Station for Plug Data butseeing that this issue is still prevalent after several PlugData updates, I;m having serious second thoughts of develoiping further for this environment.

**Comment by pereshaped** [comment]
So after weeks of scratching my head on this (no documentation or reference to this, that is searchable) it works out that once you have created your [param param1] object on the patch you are developing. Instantiate it in the right hand automation panel by adding a new param and it should automatically link to that param param1 object. To save the right hand side panel params in the automation panel, go to file and hit "export workspace" and lo and behold it loads those automation setting back again. Seeing that automation is probably one of the most core features of any plugin system in a DAW, it is really unfortunate that this vital piece of information is missing from the documentation. The only caveat is that there is no import/export workspace in the iOs AU plugin mode.
